app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -54,13 +59,125 @@ class ProfileForm extends Component {
     const { name, value } = e.target;
 
     currentFormState[name] = value;
+    this.setState({
+      formValues: currentFormState,
+      canSkip: this.formIsEmpty(currentFormState),
+    });
+  }
+
+  handleColorPickerChange(e) {
+    const { formValues } = { ...this.state };
+    const currentFormState = formValues;
+
+    const field = e.target;
+    const { name, value } = field;
+
+    let sibling = field.nextElementSibling
+      ? field.nextElementSibling
+      : field.previousElementSibling;
+    sibling.value = value;
 
+    currentFormState[name] = value;
+    this.setState({
+      formValues: currentFormState,
+      canSkip: this.formIsEmpty(currentFormState),
+    });
+  }
+
+  formIsEmpty(currentFormState) {
     // Once we've derived the new form values, check if the form is empty
     // and use that value to set the `canSkip` property on the state.
-    const formIsEmpty =
-      Object.values(currentFormState).filter((v) => v.length > 0).length === 0;
+    Object.values(currentFormState).filter((v) => v.length > 0).length === 0;
+  }
 
-    this.setState({ formValues: currentFormState, canSkip: formIsEmpty });
+  checkboxField(field) {
+    return (
+      <div class="crayons-field crayons-field--checkbox">
+        <input
+          class="crayons-checkbox"
+          type="checkbox"
+          name={field.attribute_name}
+          id={field.attribute_name}
+          onChange={this.handleChange}
+        />
+        <label class="crayons-field__label" htmlFor={field.attribute_name}>
+          {field.label}
+        </label>
+        {field.description && (
+          <p class="crayons-field__description">{field.description}</p>
+        )}
+      </div>
```

Comment:
refactor (blocking): Use the [FormField](https://storybook.forem.com/?path=/story/3-components-form-components-form-field--radio-variant) design system component instead.

```suggestion
      <FormField variant="checkbox">
        <input
          class="crayons-checkbox"
          type="checkbox"
          name={field.attribute_name}
          id={field.attribute_name}
          onChange={this.handleChange}
        />
        <label class="crayons-field__label" htmlFor={field.attribute_name}>
          {field.label}
        </label>
        {field.description && (
          <p class="crayons-field__description">{field.description}</p>
        )}
      </FormField>
```

Reply:
✅